framework/source/class/qx/log/Logger.js

```diff
@@ -118,16 +126,19 @@ qx.Bootstrap.define("qx.log.Logger",
 
       // Register appender
       var id = this.__id++;
-      this.__appender[id] = appender;
+      this.__appenders[id] = appender;
+      this.__appendersByName[appender.classname] = appender;
       appender.$$id = id;
       var levels = this.__levels;
 
       // Insert previous messages
       var entries = this.__buffer.getAllLogEvents();
       for (var i=0, l=entries.length; i<l; i++) {
-        if (levels[entries[i].level] >= levels[this.__level]) {
-          appender.process(entries[i]);
-        }
+        var entry = entries[i];
+
+        var appenders = this.__getAppenders(entry.loggerName, entry.level);
+        if (qx.lang.Array.contains(appenders, appender))
+          appender.process(entry);
       }
```

Comment:
Wouldn't it make more sense to use a hash here instead of doing a linear search every time ?